### DISCUSSION TITLE
How can I compute the probability $P(X > value)$?

### DISCUSSION THREAD

**Discussion by nucflash** [georgebv/pyextremes#88] [Q&A]
Hi, thank you for the great effort to bring Extreme Value Analysis to Python! It is extremely useful to have something like this in Python as it plays so well with all other data science tools that are around. 

I'm following [Fawcett's tutorial/lectures](http://www.mas.ncl.ac.uk/~nlf8/teaching/mas8391/background/slides2.pdf), and in one of [his example websites](https://keithnewman.shinyapps.io/extreme-explorer/) he shows an example of "Choose a Wave height to find the probability of exceeding it every year", this translates to calculating $P(X > value)$, where $value$ is the height of a wave in feet. 

I'm trying to figure out how to achieve the same in `pyextremes`. I looked at the tutorials and the code but I can't figure out a way, but I'm sure that one must exist. Can you please show me how? 

Thanks again for the great work!

<img width="613" alt="Screenshot 2024-08-01 at 9 27 39 PM" src="https://github.com/user-attachments/assets/435f50f6-a7ef-447b-bd74-05dc02bfd874">

**Comment by nucflash** [comment]
I think I found it, I think it is: 
```
1 - model.model.cdf(value)
```

CDF returns $P(X < value)$ so if I want the opposite I ask for $1 - P(X < value) = P(X > value)$

My next question is: 

Is it possible to calculate this probability within a specific return period? The question I want to answer is "What is the probability of achieving `value` in the first year, the second year, etc.". 

Thanks!

**Comment by georgebv** [comment]
You are correct. Also don't forget to factor in rate of extremes when converting return period into probability: https://github.com/georgebv/pyextremes/blob/3706ff8303027499001f0bdff5dcbe6082399a58/src/pyextremes/eva.py#L1230-L1259

Regarding your second question, extreme events are assumed to be random IID (Independent and Identically Distributed) so it doesn't matter which year you look at. The question you can answer though is `What is the probability of wave height exceeding X at least once in N years?`. That would be inverse of probability of not observing it N years in a row.

**Reply by nucflash** [reply]
Thank you George!

I see, I should divide the wave height by the `extreme_rates`. I use BM, and I use the same block_size as the return_period_size, I think I shouldn't be affected too much by it, but it is very good to know about it! 

With this in mind and for my first question, "What is the probability of wave height exceeding X every year", I write:
```
wave_height = 7  # in feet
return_period = 0.01  # ~1yr in logscale
extreme_rates = return_period_size / self.extremes_kwargs["block_size"]   # 365/365 = 1
1 - model.model.cdf(wave_height / extreme_rates)
```

And thank you for the pointer for my second question: "What is the probability of wave height exceeding X at least once in N years?". It feels like I should work along the lines of $1 - \prod_{n} P(X < x)$.
